builtin/providers/aws/resource_aws_route_test.go

```diff
@@ -0,0 +1,298 @@
+		}
+		
+		name := "aws_internet_gateway.foo"
+		gwres, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s\n", name)
+		}
+
+		if *route.GatewayId != gwres.Primary.ID {
+			return fmt.Errorf("Internet Gateway Id (Expected=%s, Actual=%s)\n", gwres.Primary.ID, *route.GatewayId)
+		}
+		
+		return nil
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRouteDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSRouteBasicConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRouteExists("aws_route.bar", &route),
+					testCheck,
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSRoute_changeCidr(t *testing.T) {
+	var route ec2.Route
+	var routeTable ec2.RouteTable
+
+	//aws creates a default route
+	testCheck := func(s *terraform.State) error {
+		if *route.DestinationCidrBlock != "10.3.0.0/16" {
+			return fmt.Errorf("Destination Cidr (Expected=%s, Actual=%s)\n", "10.3.0.0/16", *route.DestinationCidrBlock)
+		}
+
+		name := "aws_internet_gateway.foo"
+		gwres, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s\n", name)
+		}
+
+		if *route.GatewayId != gwres.Primary.ID {
+			return fmt.Errorf("Internet Gateway Id (Expected=%s, Actual=%s)\n", gwres.Primary.ID, *route.GatewayId)
+		}
+
+		return nil
+	}
+
+	testCheckChange := func(s *terraform.State) error {
+		if *route.DestinationCidrBlock != "10.2.0.0/16" {
+			return fmt.Errorf("Destination Cidr (Expected=%s, Actual=%s)\n", "10.2.0.0/16", *route.DestinationCidrBlock)
+		}
+
+		name := "aws_internet_gateway.foo"
+		gwres, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s\n", name)
+		}
+
+		if *route.GatewayId != gwres.Primary.ID {
+			return fmt.Errorf("Internet Gateway Id (Expected=%s, Actual=%s)\n", gwres.Primary.ID, *route.GatewayId)
+		}
+
+		if rtlen := len(routeTable.Routes); rtlen != 2 {
+			return fmt.Errorf("Route Table has too many routes (Expected=%d, Actual=%d)\n", rtlen, 2)
+		}
+
+		return nil
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRouteDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSRouteBasicConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRouteExists("aws_route.bar", &route),
+					testCheck,
+				),
+			},
+			resource.TestStep{
+				Config: testAccAWSRouteBasicConfigChangeCidr,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRouteExists("aws_route.bar", &route),
+					testAccCheckRouteTableExists("aws_route_table.foo", &routeTable),
+					testCheckChange,
+				),
+			},
+		},
+	})
+}
+
+// Acceptance test if mixed inline and external routes are implemented
```

Comment:
I initially meant to test (and implement) mixed inline and external routes, but decided to hold off since it's only nice to have and not necessary.  Should I remove this test entirely from the source?